tests/integration/test_rootcheck/test_rootcheck.py

```diff
@@ -0,0 +1,201 @@
+    conn = sqlite3.connect(agent_db_path)
+    cursor = conn.cursor()
+    cursor.execute("select * from pm_event")
+    rows = cursor.fetchall()
+    cursor.close()
+    conn.close()
+    return rows
+
+
+def create_injectors(agents):
+    injectors = []
+    sender = Sender(SERVER_ADDRESS, protocol=PROTOCOL)
+    for index, agent in enumerate(agents):
+        injector = Injector(sender, agent)
+        injectors.append(injector)
+        injector.run()
+        if PROTOCOL == "tcp":
+            sender = Sender(manager_address=SERVER_ADDRESS, protocol=PROTOCOL)
+    return injectors
+
+
+def send_delete_table_request(agent_id):
+    controller = SocketController(WDB_PATH)
+    controller.send(f'agent {agent_id} rootcheck delete', size=True)
+    response = controller.receive(size=True)
+    return response
+
+
+def test_rootcheck(get_configuration, configure_environment, restart_service,
+                   clean_alert_logs):
+    metadata = get_configuration.get('metadata')
+    agents_number = metadata['agents_number']
+    check_updates = metadata['check_updates']
+    check_delete = metadata['check_delete']
+
+    agents = create_agents(agents_number, SERVER_ADDRESS, CRYPTO)
+
+    injectors = create_injectors(agents)
+
+    # Let rootcheck events to be sent for 60 seconds
+    time.sleep(60)
+
+    for injector in injectors:
+        injector.stop_receive()
+
+    # Service needs to be stopped otherwise db lock will be held by Wazuh db
+    control_service('stop')
+
+    # Check that logs have been added to the sql database
+    for agent in agents:
+        rows = retrieve_rootcheck_rows(agent.id)
+        db_string = [row[3] for row in rows]
+
+        logs_string = [':'.join(x.split(':')[2:]) for x in
+                       agent.rootcheck.rootcheck]
+        for log in logs_string:
+            assert log in db_string, f"Log: \"{log}\" not found in Database"
+
+        alerts_description = None
+        with open(ALERTS_JSON_PATH, 'r') as f:
+            json_lines = [json.loads(x) for x in f.readlines()]
+            alerts_description = [x['full_log'] for x in json_lines
+                                  if 'rootcheck' in x['decoder']['name']]
+            for log in logs_string:
+                if log not in ['Starting rootcheck scan.',
+                               'Ending rootcheck scan.']:
+                    assert log in alerts_description, f"Log: \"{log}\" " \
+                            "not found in alerts file"
+
+    if check_updates:
+        # Service needs to be stopped otherwise db lock will be held by
+        # Wazuh db
+        control_service('start')
+
+        update_threshold = time.time()
+
+        injectors = create_injectors(agents)
+
+        # Let rootcheck events to be sent for 60 seconds
+        time.sleep(60)
+
+        # Check that logs have been updated
+        for agent in agents:
+            rows = retrieve_rootcheck_rows(agent.id)
+
+            db_string = [row[3] for row in rows]
```

Comment:
Unused variable. Perhaps some verification with this variable is missing. Otherwise it would be better to remove it.
```suggestion
```